drivers/i2c/lsm9ds1.py

```diff
@@ -0,0 +1,164 @@
+>>> lsm.read_magnet()     # (x,y,z) in gauss 
+(0.5358887, -0.001586914, -0.1228027)
+>>> for g,a in lsm.iter_accel_gyro(): print(g,a)    # using fifo
+"""
+import array
+
+class LSM9DS1:
+    WHO_AM_I = const(0xf)
+    CTRL_REG1_G = const(0x10)
+    INT_GEN_SRC_G = const(0x14)
+    OUT_TEMP = const(0x15)  
+    OUT_G = const(0x18) 
+    CTRL_REG4_G = const(0x1e)
+    
+    STATUS_REG = const(0x27)
+    OUT_XL = const(0x28)
+    FIFO_CTRL_REG = const(0x2e)
+    FIFO_SRC = const(0x2f)
+    
+    OFFSET_REG_X_M = const(0x05)
+    CTRL_REG1_M = const(0x20)
+    OUT_M = const(0x28)
+    
+    SCALE_GYRO = [(245,0),(500,1),(2000,3)]
+    SCALE_ACCEL = [(2,0),(4,2),(8,3),(16,1)]
+    
+    def __init__(self, i2c, address_gyro=106, address_magnet=28):
+        self.i2c = i2c
+        self.address_gyro = address_gyro
+        self.address_magnet = address_magnet
+        # check id's of accelerometer/gyro and magnetometer
+        if (self.read_id_magnet() != b'=') or (self.read_id_gyro() != b'h'):
+            raise OSError("Invalid LSM9DS1 device, using address {}/{}".format(
+                    address_gyro,address_magnet))
+        # allocate scratch buffer for efficient conversions and memread op's
+        self.scratch = array.array('B',[0,0,0,0,0,0])
+        self.scratch_int = array.array('h',[0,0,0])
+        self.init_gyro_accel()
+        self.init_magnetometer()
+        
+    def init_gyro_accel(self, sample_rate=6, scale_gyro=0, scale_accel=0):
+        """ Initalizes Gyro and Accelerator.
+        sample rate: 0-6 (off, 14.9Hz, 59.5Hz, 119Hz, 238Hz, 476Hz, 952Hz)
+        scale_gyro: 0-2 (245dps, 500dps, 2000dps ) 
+        scale_accel: 0-3 (+/-2g, +/-4g, +/-8g, +-16g)
+        """
+        assert sample_rate <= 6, "invalid sampling rate: %d" % sample_rate
+        assert scale_gyro <= 2, "invalid gyro scaling: %d" % scale_gyro
+        assert scale_accel <= 3, "invalid accelerometer scaling: %d" % scale_accel
+        
+        i2c = self.i2c
+        addr = self.address_gyro
+        mv = memoryview(self.scratch)        
+        # angular control registers 1-3 / Orientation
+        mv[0] = ((sample_rate & 0x07) << 5) | ((self.SCALE_GYRO[scale_gyro][1] & 0x3) << 3) 
+        mv[1:4] = b'\x00\x00\x00'
+        i2c.mem_write(mv[:5], addr, CTRL_REG1_G)
+        # ctrl4 - enable x,y,z, outputs, no irq latching, no 4D
+        # ctrl5 - enable all axes, no decimation
+        # ctrl6 - set scaling and sample rate of accel 
+        # ctrl7,8 - leave at default values
+        # ctrl9 - FIFO enabled
+        mv[0] = mv[1] = 0x38
+        mv[2] = ((sample_rate & 7) << 5) | ((self.SCALE_ACCEL[scale_accel][1] & 0x3) << 3)
+        mv[3] = 0x00
+        mv[4] = 0x4
+        mv[5] = 0x2
+        i2c.mem_write(mv[:6], addr, CTRL_REG4_G )
+        
+        # fifo: use continous mode (overwrite old data if overflow)
+        i2c.mem_write(0x00, addr, FIFO_CTRL_REG)
+        i2c.mem_write(6 << 5, addr, FIFO_CTRL_REG)
```

Comment:
I see you are using the old `pyb` API here. Would it make sense to use the `machine` API instead, so that it would work on newer ports too?